pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -278,6 +599,26 @@ func (r *repository) getImageStreamImage(dgst digest.Digest) (*imageapi.ImageStr
 	return r.registryClient.ImageStreamImages(r.namespace).Get(r.name, dgst.String())
 }
 
+// manifestFromImageWithCachedLayers loads the image and then caches any located layers
+func (r *repository) manifestFromImageWithCachedLayers(image *imageapi.Image, cacheName string) (*schema1.SignedManifest, error) {
+	manifest, err := r.manifestFromImage(image)
```

Comment:
`manifesFromImage` is called only by `manifestFromImageWithCachedLayers`. Perhaps they could be merged under the shorter name.